include/RAJA/pattern/params/reducer.hpp

```diff
@@ -83,16 +55,38 @@ namespace detail
   // Basic Reducer
   //
   //
-  template <typename Op, typename T>
+
+  // Basic data type Reducer
+  // T must be a basic data type
+  // IndexType not used here
+  // VType must be ValOp<T, Op>
+  template <typename Op, typename T, typename VType>
   struct Reducer : public ForallParamBase {
-    using op = Op;
-    using value_type = T;
+    //using op = Op<T,T,T>;
+    using value_type = T; // This is a basic data type
```

Comment:
TTe set of aliases should be as consistent as possible between this Reduce and the partial specialization.